rich/rule.py

```diff
@@ -21,29 +21,37 @@ def __init__(
         self,
         title: Union[str, Text] = "",
         *,
-        character: str = None,
+        characters: str = "─",
         style: Union[str, Style] = "rule.line",
         end: str = "\n",
     ) -> None:
-        if character and cell_len(character) != 1:
-            raise ValueError("'character' argument must have a cell width of 1")
+        if cell_len(characters) < 1:
+            raise ValueError(
+                "'characters' argument must have at least a cell width of 1"
+            )
         self.title = title
-        self.character = character
+        self.characters = characters
         self.style = style
         self.end = end
 
     def __repr__(self) -> str:
-        return f"Rule({self.title!r}, {self.character!r})"
+        return f"Rule({self.title!r}, {self.characters!r})"
 
     def __rich_console__(
         self, console: Console, options: ConsoleOptions
     ) -> RenderResult:
         width = options.max_width
 
-        character = self.character or "─"
+        characters = self.characters or "─"
 
+        if cell_len(characters) == 1:
+            chars_len = get_char_size(characters)
+        else:
+            chars_len = 0
+            for i in list(characters):
+                chars_len += get_char_size(i)
         if not self.title:
-            yield Text(character * width, self.style)
+            yield Text(characters * (width // chars_len), self.style)
```

Comment:
This may not extend to the end if characters is more than one character.

For instance if the terminal width is 100, and characters is "abc" the rule will be 99 characters since (100 // 3) is 33.

Suggest calculating the characters by adding one extra repetition then truncating to width.

Something like this:

```python
text = Text(characters * (width // chars_len) + 1, self.style) 
text.truncate(width)
```

Reply:
Needed extra parenthesis `((width // chars_len) + 1)` done